app/src/main/java/de/rampro/activitydiary/ui/history/HistoryActivity.java

```diff
@@ -403,4 +441,60 @@ public void addDetailAdapter(long diaryEntryId, DetailRecyclerViewAdapter adapte
 
     }
 
+    /** Checks date format and also checks date can be parsed (used for not existing dates like 35.13.2000)
+     * @param date input that is checked
+     * @param dateFormat format under we want to check (parse) entered date
+     */
+    private boolean checkDateFormat(String date, String dateFormat){
+        if (dateFormat == null)
+            return false;
+
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dateFormat);
+        simpleDateFormat.setLenient(false);
+        switch (dateFormat) {
```

Comment:
you handle here only 2 specific formats... and actually it is even handled elsewhere. It would be better to do the check and the real parsing in the same method, to simplify a change (add a new supported format). - if it is split, several locations would need to be adjusted. Also we should not try to do the date parsing ourselves, the libraries can do than better and more generic. -> I propose to support ISO format "YYYY-MM-DD" and default format of the users locale...